### PR TITLE
Compensating the lack of normative IMA references via appendix text and commit specific pointers to the Linux kernel

### PR DESCRIPTION
**Pull request by henkbirkholz**

### CODE REVIEW

File: draft-ietf-rats-yang-tpm-charra.md [hunk 1]
```diff
@@ -204,23 +204,27 @@ normative:
 
   bios-log:
     target: https://trustedcomputinggroup.org/wp-content/uploads/PC-ClientSpecific_Platform_Profile_for_TPM_2p0_Systems_v51.pdf
-    title: "TCG PC Client Platform Firmware Profile Specification, Section 9.4.5.2" 
+    title: "TCG PC Client Platform Firmware Profile Specification, Section 9.4.5.2"
 
   ima-log:
```
Comment (kaduk) [inline draft-ietf-rats-yang-tpm-charra.md:209]: If IMA is a linux thing, do we want to continue to use the name "ima-log" for a link to the TCG document?

File: draft-ietf-rats-yang-tpm-charra.md [hunk 2]
```diff
@@ -514,3 +518,44 @@ For the YANG module ietf-tcg-algs.yang, please use care when selecting specific
...
+
+In support of the appraisal of Evidence, IMA maintains an ordered list of measurements in kernel-space, the Stored Measurement Log (SML), for all files that have been measured before execution since the operating system was started.
+Although IMA can be used without a TPM, it is typically used in conjunction with a TPM to anchor the integrity of the SML in a hardware-protected secure storage location, i.e., Platform Configuration Registers (PCRs) provided by TPMs.
+IMA provides the SML in both binary and ASCII representations in the Linux security file system *securityfs* (`/sys/kernel/security/ima/`).
+
+IMA templates define the format of the SML, i.e., which fields are included in a log record.
+Examples are file path, file hash, user ID, group ID, file signature, and extended file attributes.
+IMA comes with a set of predefined template formats and also allows a custom format, i.e., a format consisting of template fields supported by IMA.
+Template usage is typically determined by boot arguments passed to the kernel.
+Alternatively, the format can also be hard-compiled into custom kernels.
```
Comment (kaduk) [inline draft-ietf-rats-yang-tpm-charra.md:537]: "hard-compiled" is not a phrase I see often; "hard-coded" or just "compiled" are more common.  (Also appears again later.)
(Is it "wrong"?  Probably not.)

File: draft-ietf-rats-yang-tpm-charra.md [hunk 3]
```diff
@@ -514,3 +518,44 @@ For the YANG module ietf-tcg-algs.yang, please use care when selecting specific
...
+IMA policies define which files are measured using the IMA policy language.
+Built-in policies can be passed as boot arguments to the kernel.
+Custom IMA policies can be defined once during runtime or be hard-compiled into a custom kernel.
+If no policy is defined, no measurements are taken and IMA is effectively disabled.
+
+# IMA for Network Equipment Boot Logs {#netequip-boot-log}
+
+Network equipment can generally implement similar IMA-protected functions to generate measurements (Claims) about the boot process of a device and enable corresponding remote attestation.
+Network Equipment Boot Logs combine the measurement and logging of boot components and operating system components (executables and files) into a single log file in identical IMA format.
+<!-- Changes to the boot components are required to allow them to log in IMA format. -->
```
Comment (kaduk) [inline draft-ietf-rats-yang-tpm-charra.md:549]:
```suggestion
Note that the format used for logging measurement of boot components in this scheme differs from the boot logging strategy described elsewhere in this document.
```
(A section reference is probably better than "elsewhere in this document" but I don't have one handy.)

File: draft-ietf-rats-yang-tpm-charra.md [hunk 4]
```diff
@@ -514,3 +518,44 @@ For the YANG module ietf-tcg-algs.yang, please use care when selecting specific
...
+
+During the boot process of the network device, i.e., from BIOS to the end of the operating system and user-space, all files executed during this process can be measured and logged in the order of their execution.
```
Comment (kaduk) [inline draft-ietf-rats-yang-tpm-charra.md:551]:
```suggestion
During the boot process of the network device, i.e., from BIOS to the end of the operating system and user-space, all files executed can be measured and logged in the order of their execution.
```

File: draft-ietf-rats-yang-tpm-charra.md [hunk 5]
```diff
@@ -514,3 +518,44 @@ For the YANG module ietf-tcg-algs.yang, please use care when selecting specific
...
+When the Verifier initiates a remote attestation process (e.g., challenge-response remote attestation as defined in this document), the network equipment takes on the role of an Attester and can convey to the Verifier Claims that comprise the measurement log as well as the corresponding PCR values (Evidence) of a TPM.
+
+The verifier can appraise the integrity (compliance with the Reference Values) of each executed file by comparing its measured value with the Reference Value.
+Based on the execution order, the Verifier can compute a PCR reference value (by replaying the log) and compare it to the Measurement Log Claims obtained in conjunction with the PCR Evidence to assess their trustworthiness with respect to an intended operational state.
+
+Not only during the operating system loading phase, even during the BIOS boot phase, network equipment usually executes multiple components.
```
Comment (kaduk) [inline draft-ietf-rats-yang-tpm-charra.md:557]:
```suggestion
Network equipment usually executes multiple components in parallel.  This holds not only during the operating system loading phase, but also even during the BIOS boot phase.
```

File: draft-ietf-rats-yang-tpm-charra.md [hunk 6]
```diff
@@ -514,3 +518,44 @@ For the YANG module ietf-tcg-algs.yang, please use care when selecting specific
...
+With this measurement log mechanism, network equipment can take on the role of an Attester, proving to the Verifier the trustworthiness of its boot process.
+Using the measurement log, Verifiers can precisely identify mismatching log entries to infer potentially tampered components.
+
+This mechanism also supports scenarios that modify files on the Attester and are executed during the boot phase (e.g., updating/patching) by simply updating the appropriate Reference Values in Reference Integrity Manifests that inform Verifiers about how an Attester is composed.
```
Comment (kaduk) [inline draft-ietf-rats-yang-tpm-charra.md:561]:
```suggestion
This mechanism also supports scenarios that modify files on the Attester that are subsequently executed during the boot phase (e.g., updating/patching) by simply updating the appropriate Reference Values in Reference Integrity Manifests that inform Verifiers about how an Attester is composed.
```

**Review by kaduk — APPROVED** [review]
Thanks, this does a good job bridging the gap between what we say in the body of the document and what is available in external references.

File: draft-ietf-rats-yang-tpm-charra.md (hunk 1 above)
Comment (henkbirkholz) [reply draft-ietf-rats-yang-tpm-charra.md:209]: yes, the "TCG doc" is referencing it in a comprehensive manner and while it is a indirection, I think i it is a useful one.

File: draft-ietf-rats-yang-tpm-charra.md (hunk 2 above)
Comment (henkbirkholz) [inline draft-ietf-rats-yang-tpm-charra.md:537]:
```suggestion
Alternatively, the format can also be hard-coded into custom kernels.
```

Yeah

File: draft-ietf-rats-yang-tpm-charra.md (hunk 3 above)
Comment (henkbirkholz) [reply draft-ietf-rats-yang-tpm-charra.md]: Thanks! As you can see, we were not entirely certain about that. That helped!

File: draft-ietf-rats-yang-tpm-charra.md (hunk 5 above)
Comment (henkbirkholz) [reply draft-ietf-rats-yang-tpm-charra.md]: Every simplification is welcome. Accepted.

File: draft-ietf-rats-yang-tpm-charra.md (hunk 6 above)
Comment (henkbirkholz) [reply draft-ietf-rats-yang-tpm-charra.md]: yes

**Comment by henkbirkholz** [comment]
Self-approved

**Pull request merged**